saas/tests/test_saas.py

```diff
@@ -56,49 +43,7 @@ def assert_no_error_in_db(self, dbname):
     @classmethod
     def setUpClass(cls):
         super(TestSaas, cls).setUpClass()
-        cls.env = cls.env(context=dict(
-            cls.env.context,
-            test_queue_job_no_delay=True,
-        ))
-
-        cls.saas_template_1 = cls.env['saas.template'].create({
-            'template_module_ids': [(0, 0, {
-                'name': MODULE_TO_INSTALL,
-            })],
-            'template_post_init': 'env[\'mail.message\'].create({\'subject\': \'' + TEMPLATE_TEST_SUBJECT + '\', })',
-            'build_post_init': DEFAULT_BUILD_PYTHON_CODE + 'env[\'{mail_message}\'].create({{\'subject\': \'' + BUILD_TEST_SUBJECT + '\', }})',
-        })
-
-        cls.saas_template_2 = cls.env['saas.template'].create({
-            'template_module_ids': [(0, 0, {
-                'name': 'mail',
-            })],
-            'template_post_init': 'env[\'mail.message\'].create({\'subject\': \'' + TEMPLATE_TEST_SUBJECT + '\', })',
-        })
-
-        cls.saas_operator_1 = cls.env['saas.operator'].create({
-            'type': 'local',
-            'db_url_template': 'http://{db_name}.{db_id}.127.0.0.1.nip.io:8069',
-            'master_url': 'http://saas.127.0.0.1.nip.io:8069',
-        })
-
-        cls.saas_operator_2 = cls.env['saas.operator'].create({
-            'type': 'local',
-            'db_url_template': 'http://{db_name}.{db_id}.127.0.0.1.nip.io:8069',
-            'master_url': 'http://saas.127.0.0.1.nip.io:8069',
-        })
-
-        cls.saas_template_operator_1 = cls.env['saas.template.operator'].create({
-            'template_id': cls.saas_template_1.id,
-            'operator_id': cls.saas_operator_1.id,
-            'operator_db_name': DB_TEMPLATE_1,
-        })
-
-        cls.saas_template_operator_2 = cls.env['saas.template.operator'].create({
-            'template_id': cls.saas_template_2.id,
-            'operator_id': cls.saas_operator_2.id,
-            'operator_db_name': DB_TEMPLATE_2,
-        })
+        Common.setup_saas_env(cls)
```

Comment:
You can inherit Common and say ``self.setup_saas_env()``

Reply:
Then I will have trouble in saas' class method

Reply:
You can use multiple inheritance

Reply:
> You can inherit Common and say `self.setup_saas_env()`

I can't use `self` here

Reply:
So, actually I can use `cls.setup_saas_env(cls)`

Reply:
I need to pass `cls` because of this error `TypeError: setup_saas_env() missing 1 required positional argument: 'self'`

Reply:
Your error doesn't mean that you cannot follow my advice. How do you inherit? Do you use ``class TestSaas(SavepointCase, Common):``?

Reply:
Also, check the decorator for ``setup_saas_env``

Reply:
> I can't use self here

![image](https://user-images.githubusercontent.com/186131/60813645-9acc3880-a1ad-11e9-91de-a2e00e0aab0d.png)

Reply:
> class TestSaas(SavepointCase, Common):

For now everything works with `class TestSaas(SavepointCase, Common):`